api/scholarqa/scholar_qa.py

```diff
@@ -395,9 +400,36 @@ def answer_query(self, query: str, inline_tags: bool = True) -> Dict[str, Any]:
         task_result = self.run_qa_pipeline(tool_request, inline_tags)
         return task_result.model_dump()
 
-    def gen_table_thread(self, user_id: str, query: str, dim: Dict[str, Any],
+    def gen_table_thread(self, task_id: str, user_id: str, query: str, dim: Dict[str, Any],
```

Comment:
task_id should be available as a data member of the class
```suggestion
    task_id = self.task_id if self.task_id else self.tool_request.task_id
```